src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableEntityContext.cs

```diff
@@ -424,10 +426,14 @@ private void SignalEntityInternal(EntityId entity, DateTime? scheduledTimeUtc, s
             {
                 InstanceId = EntityId.GetSchedulerIdFromEntityId(entity),
             };
+
+            using var signalEntityActivity = TraceHelper.StartActivityForCallingOrSignalingEntity(target.InstanceId, entity.EntityName, operation, true, Activity.Current?.Context, this.InstanceId);
+            var parentTraceContext = new ActivityContext(signalEntityActivity.TraceId, signalEntityActivity.SpanId, signalEntityActivity.ActivityTraceFlags, signalEntityActivity.TraceStateString);

```

Comment:
It looks like we call this constructor with a long list of parameters in multiple places. I haven't examined the code super closely yet, but could we remove some of the redundancy a bit by creating a method that allows us to do something like `var parentTraceContext = signalEntityActivity.ToActivityContext()`?